pkg/controller/cluster/http_handlers.go

```diff
@@ -116,3 +128,210 @@ func validateBucketName(bucket string) (bool, error) {
+		API:             webIdentity,
+		TenantNamespace: tenantNamespace,
+	}
+
+	ctx := context.WithValue(r.Context(), contextLogKey, &reqInfo)
+
+	if err != nil {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue, fmt.Errorf("tenant namespace is missing"))
+	}
+
+	// Parse the incoming form data.
+	if err := xhttp.ParseForm(r); err != nil {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue, fmt.Errorf("Error parsing request: %s", err))
+		return
+	}
+
+	if r.Form.Get(stsVersion) != stsAPIVersion {
+		err := fmt.Errorf("invalid STS API version %s, expecting %s", r.Form.Get("Version"), stsAPIVersion)
+		writeSTSErrorResponse(ctx, w, true, ErrSTSMissingParameter, err)
+		return
+	}
+
+	action := r.Form.Get(stsAction)
+	switch action {
+	// For now we only do WebIdentity, leaving it in case we want to implement certificate authentication
+	case webIdentity:
+	default:
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue, fmt.Errorf("unsupported action %s", action))
+		return
+	}
+
+	token := strings.TrimSpace(r.Form.Get(stsWebIdentityToken))
+
+	if token == "" {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSMissingParameter, fmt.Errorf("missing %s", stsWebIdentityToken))
+		return
+	}
+
+	// VALIDATE JWT
+	accessToken := r.Form.Get(stsWebIdentityToken)
+	saAuthResult, err := c.ValidateServiceAccountJWT(&ctx, accessToken)
+	if err != nil {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidIdentityToken, err)
+		return
+	}
+	if !saAuthResult.Status.Authenticated {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSAccessDenied, fmt.Errorf("Access denied: Invalid Token"))
+		return
+	}
+	chunks := strings.Split(strings.Replace(saAuthResult.Status.User.Username, "system:serviceaccount:", "", -1), ":")
+	// saNamespace Service account Namespace
+	saNamespace := chunks[0]
+	// saName service account username
+	saName := chunks[1]
+
+	// Authorized PolicyBindings for the Service Account
+	policyBindings := []v1alpha1.PolicyBinding{}
+	pbs, err := c.minioClientSet.StsV1alpha1().PolicyBindings(tenantNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInternalError, fmt.Errorf("Error obtaining PolicyBindings: %s", err))
+		return
+	}
+
+	for _, pb := range pbs.Items {
+		if pb.Spec.Application.Namespace == saNamespace && pb.Spec.Application.ServiceAccount == saName {
+			policyBindings = append(policyBindings, pb)
+		}
+	}
+	if len(policyBindings) == 0 {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSAccessDenied, fmt.Errorf("Service account '%s' has no PolicyBindings in namespace '%s'", saAuthResult.Status.User.Username, tenantNamespace))
+		return
+	}
+
+	tenants, err := c.minioClientSet.MinioV2().Tenants(tenantNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil || len(tenants.Items) == 0 {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue, fmt.Errorf("No Tenants available in the namespace '%s'", tenantNamespace))
```

Comment:
there's a possible `k8serrors.IsNotFound(err)` for when the tenant is not found, and should be handled different from any other error (api rate limit) so we should handle that and return a different error

Reply:
done!
```go
	tenants, err := c.minioClientSet.MinioV2().Tenants(tenantNamespace).List(ctx, metav1.ListOptions{})

 	if err != nil || len(tenants.Items) == 0 {
 		if k8serrors.IsNotFound(err) {
 			writeSTSErrorResponse(w, true, ErrSTSInvalidParameterValue, fmt.Errorf("No tenant found in namespace '%s'", tenantNamespace))
 			return
 		}
 		writeSTSErrorResponse(w, true, ErrSTSInvalidParameterValue, fmt.Errorf("Error getting tenant in namespace '%s'", tenantNamespace))
 		return
 	}
```
https://github.com/minio/operator/pull/1414/files#diff-28f72f89bc566e0742da50838fe4a9ca64c620bcbad5aee1c9b97b1413f6a321R232-R241